lib/pbench/cli/server/report.py

```diff
@@ -368,16 +369,26 @@ def columnize(
         click.echo(line)
 
 
-def summarize_dates(rows: Iterator[Row], width: int = 80):
+def summarize_dates(query: Query, options: dict[str, Any]):
+    width: int = options.get("width")
     by_year = defaultdict(int)
     by_month = defaultdict(int)
     by_day = defaultdict(int)
     by_weekday = defaultdict(int)
     by_hour = defaultdict(int)
-
-    day = datetime.datetime.now(datetime.timezone.utc).replace(
-        hour=0, minute=0, second=0, microsecond=0
-    )
+    since = options.get("since")
+    until = options.get("until")
+
+    start = since if since else datetime.datetime.fromtimestamp(0.0)
+    end = until if until else datetime.datetime.now(datetime.timezone.utc)
+
+    # It's convenient to use `--until YYYY-MM-01` to see a month (though
+    # technically that would include a YYYY-MM-01:00:00.00 timestamp), but
+    # bucketizing the day or week based on that anomaly isn't very useful, so
+    # back up the "day" one millisecond to move it into the last day of the
+    # previous month.
+    day = end - datetime.timedelta(milliseconds=1)
+    day = day.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
1. Yes, I seriously considered changing `--until` to be a strict `<`. (In fact, I made the change, but then retracted it.) The reason I adopted the somewhat stilted names in the first place (for audit) was that they follow `journalctl` convention, which seemed (at least for audit) appropriate. That's less of a concern here, but I was reluctant to break compatibility with the audit tool and I can't quite convince myself that compatibility is a problem.
2. Backing up the top range here is because there's a semantic distinction between the filtering based on `end` and the bucketizing base on `day`. I'm bucketizing by year, month, day, week: if `end` is in the middle anyway, then backing up by a second for `day` doesn't hurt: but if `end` is a boundary like "next day", "next week", "next month", or "next year", then backing it up makes the final bucket meaningful. (And, yes, perhaps even more so with a `< end`, but to the limited extent it's possible to consider those two aspects as independent, I endeavored to do so just because I didn't want this ad hoc "non work" project to become a massive time investment.)